integration/kotlinx-coroutines-firebase-android/kotlinx-coroutines-firebase-android/src/main/java/kotlinx/coroutines/experimental/firebase/android/Task.kt

```diff
@@ -0,0 +1,55 @@
+package kotlinx.coroutines.experimental.firebase.android
+
+import com.google.android.gms.tasks.Task
+import kotlinx.coroutines.experimental.suspendCancellableCoroutine
+import kotlin.coroutines.experimental.suspendCoroutine
+
+/**
+ * Coroutine support to Firebase Task interface
+ *
+ * This method is not intended to be used in your code. You should probably use [await] sinsce it's
+ * more idiomatic.
+ *
+ * The implementation is pretty simple. It justs use a [suspendCoroutine] to encapsulate the
+ * Firebase [com.google.android.gms.tasks.OnCompleteListener] interface.
+ *
+ */
+private suspend fun <T> awaitTask(task: Task<T>): T = suspendCancellableCoroutine { continuation ->
```

Comment:
Please merge this method and `await`

Reply:
Changed in https://github.com/Kotlin/kotlinx.coroutines/pull/426/commits/efb8d1d3a81fb70601109f80eba45e11db9359de.